support/controlplane-component/status.go

```diff
@@ -0,0 +1,167 @@
+package controlplanecomponent
+
+import (
+	"fmt"
+	"strings"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	assets "github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/v2/assets"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *controlPlaneWorkload) checkDependencies(cpContext ControlPlaneContext) ([]string, error) {
+	if len(c.dependencies) == 0 {
+		return nil, nil
+	}
+
+	componentsList := &hyperv1.ControlPlaneComponentList{}
+	if err := cpContext.Client.List(cpContext, componentsList, client.InNamespace(cpContext.HCP.Namespace)); err != nil {
+		return nil, err
+	}
+
+	unavailableDependencies := sets.New(c.dependencies...)
+	for _, component := range componentsList.Items {
+		if !unavailableDependencies.Has(component.Name) {
+			continue
+		}
+
+		availableCondition := meta.FindStatusCondition(component.Status.Conditions, string(hyperv1.ControlPlaneComponentAvailable))
```

Comment:
I don't think it's enough to check that the dependency is available. It should also match the desired version. That way reconcile during an upgrade can occur in a predetermined order. First etcd, then kas, etc. All of them will be available, but not necessarily at the desired version.

Reply:
makes sense.